src/inmanta/data/__init__.py

```diff
@@ -5494,8 +5494,10 @@ async def update_persistent_state(
             "last_deployed_version": last_deployed_version,
         }
         query_parts = [f"{k}={args(v)}" for k, v in invalues.items() if v is not None]
-        if deployment_result:
-            query_parts.append(f"deployment_result={args(deployment_result.name)}")
+        if state:
+            query_parts.append(f"deployment_result={args(state.deployment_result.name)}")
+            # TODO: split blocked status field to make raceless
```

Comment:
There are comments the blocked field should only be updated under lock as both release and deplot path write it. 

We could get rid of that by splitting this field into 'blocked' and 'blocked_transient' booleans. These encode the 3 states without write conflict, then again, it may not be worth it, as the lock is available. .